Mathlib/MeasureTheory/Measure/Tight.lean

```diff
@@ -0,0 +1,469 @@
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne, Josha Dekker
+-/
+import Mathlib.MeasureTheory.Measure.Regular
+import Mathlib.Topology.Metrizable.Uniformity
+import Mathlib.Analysis.SpecificLimits.Basic
+
+/-!
+# (Pre-)tight measures
+The key definition of interest here is that of tight measures, `IsTight`. We first introduce two
+weaker notions, `IsSeparable` and `IsPretight`, which are equivalent on complete metric spaces. We
+provide some basic API for these notions and prove Ulam's tightness theorem
+(`of_isSeparableSpace_completeSpace`) and its strengthening `of_isSeparable_completeSpace`.
+
+## Main definitions
+* `IsSeparable`: A measure `μ` is separable if there is a separable set `S` such that
+  `μ S = μ Set.univ`.
+* `IsPretight`: A measure `μ` is pre-tight if for all `0 < ε`, there exists `K` totally bounded such
+  that `μ Kᶜ ≤ ε`.
+* `IsTight`: A measure `μ` is tight if for all `0 < ε`, there exists `K` compact such that
+  `μ Kᶜ ≤ ε`. This is defined in terms of filters. `IsTight.iff_compact_sets` establishes
+  equivalence with the usual definition.
+
+## Main statements
+
+* `of_isSeparableSpace_completeSpace`: Ulam's tightness theorem: a finite measure on a complete
+  separable metric space is tight.
+* `of_isSeparable_completeSpace`: A strengthening of Ulam's tightness theorem: a finite,
+  separable measure on a complete metric space is tight.
+* `of_innerRegular`: Every finite, inner-regular measure on a T2 space is tight.
+
+## Notation
+
+
+## Implementation details
+
+
+-/
+open Topology Filter Uniformity Uniform MeasureTheory Set
+open scoped ENNReal
+
+namespace MeasureTheory
+
+variable {α ι : Type*}
+
+variable [MeasurableSpace α] {μ : Measure α}
+
+/-- For a finite measure `μ`, we can extract from a countable cover that has full measure, a finite
+cover of accumulated sets that has `ε`-almost full measure. -/
+private lemma almost_cover_has_approx_accumulate [MeasurableSpace α] {μ : Measure α}
+    [IsFiniteMeasure μ] (K : ℕ → Set α) (h : μ (⋃ n, K n) = μ Set.univ) :
+    ∀ ε > 0, ∃ n, μ Set.univ - ε ≤ μ (Set.Accumulate K n) := by
+  rintro ε hε
+  have : Filter.Tendsto (μ ∘ Set.Accumulate K) Filter.atTop (nhds (μ (⋃ n, Set.Accumulate K n))) :=
+    MeasureTheory.tendsto_measure_iUnion Set.monotone_accumulate
+  rw [ENNReal.tendsto_atTop (measure_ne_top μ (⋃ n, Accumulate K n)), Set.iUnion_accumulate] at this
+  obtain ⟨N, hN⟩ := this ε hε
+  use N
+  simp_all only [Function.comp_apply, mem_Icc, le_refl]
+
+/-- For a finite measure `μ`, we can extract from a countable cover that has full measure, a finite
+cover of accumulated sets for which the complement has measure below `ε`. -/
+private lemma almost_cover_has_approx_accumulate_compl [MeasurableSpace α] [TopologicalSpace α]
+    [OpensMeasurableSpace α] {μ : Measure α} [IsFiniteMeasure μ] (K : ℕ → Set α)
+    (hKclosed : ∀ n, IsClosed (K n)) (h : μ (⋃ n, K n) = μ Set.univ) :
```

Comment:
The relevant property here is no that the `K n` are closed, but that they are measurable. Could you switch the lemma to this assumption?